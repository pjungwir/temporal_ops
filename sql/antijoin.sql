SELECT  a.id, j.id, COALESCE(j.valid_at, a.valid_at) AS valid_at
FROM    a
LEFT JOIN LATERAL (
  SELECT  b.id, UNNEST(multirange(a.valid_at) - range_agg(b.valid_at)) AS valid_at
  FROM    b
  WHERE   a.id = b.id
  AND     a.valid_at && b.valid_at
  GROUP BY b.id
) AS j ON true;
