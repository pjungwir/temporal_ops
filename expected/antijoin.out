SELECT  a.id, j.id, COALESCE(j.valid_at, a.valid_at) AS valid_at
FROM    a
LEFT JOIN LATERAL (
  SELECT  b.id, UNNEST(multirange(a.valid_at) - range_agg(b.valid_at)) AS valid_at
  FROM    b
  WHERE   a.id = b.id
  AND     a.valid_at && b.valid_at
  GROUP BY b.id
) AS j ON true;
 id | id | valid_at 
----+----+----------
  1 |  1 | [1,5)
  1 |  1 | [10,15)
  2 |    | [1,20)
  4 |    | [1,20)
(4 rows)

